### PR TITLE
Remote logging

### PR DESCRIPTION
**Pull request by ThanatosGit**
Connects to the new exlaunch.

~~**TODO: Logs should go into the window and not into console**~~

### CODE REVIEW

File: src/remote_dread_lua_console/lua_executor.py [hunk 1]
```diff
@@ -60,6 +84,78 @@ def ip(self, value):
...
+        return await asyncio.wait_for(self._socket.reader.read(1), timeout)
+
+    async def check_header(self):
+        received_number: bytes = await asyncio.wait_for(self._socket.reader.read(1), None)
+        if received_number[0] != self._socket.request_number:
+            raise RuntimeError(f"Expected response {self._socket.request_number}, got {received_number}")
+
+    async def parse_packet(self, packet_type):
+        response = None
+        match packet_type[0]:
```
Comment (henriquegemignani) [inline src/remote_dread_lua_console/lua_executor.py:107]: oh hey this feature. I keep forgetting it's a thing lol

**Comment by henriquegemignani** [comment]
Is there anything else left for this?

**Comment by ThanatosGit** [comment]
1. I need to update the comment about the packet structure.
2. The log goes via Signal/Slot from the executor to the console window. With lua_log the game spams like 2500 messages if you start a new game and it needs like 1 min 10 secs to pass this via signal/slot. Do you know a better way?

**Comment by henriquegemignani** [comment]
If Signal/Slot is that slow, you could add a method that does:
- adds the new message to a "pending messages" list
- emits the signal if the list was previously empty

And then you connect that signal to a function that adds everything at once.
I'd keep anything that changes UI elements to being called from a slot, though since nothing here actually uses threads it wouldn't even be needed

**Comment by ThanatosGit** [comment]
Ok, I tried that. No change in performance. It is actually the `QListWidget` with poor performance. See https://stackoverflow.com/questions/7103943/qt-qlistwidget-performance-issues

With `self.history_widget.setUniformItemSizes(True)` it's much faster but ofc no horizontal scroll bar for long lines. Maybe we can use a different widget

**Comment by henriquegemignani** [comment]
Maybe with a QListView instead of QListWidget?
You can also try merging multiple log lines into a single item of the list.

**Comment by ThanatosGit** [comment]
Final conclusion:
I patched it with what you said first:
- pending list
- emit the signal if it was previously empty

But I emit the signal now with a delay of 200 ms, so it can collect more messages. This is because I initially thought the Slot/Signal was the problem but it is actually the rendering of the QListWidget. So, with a delay the pending list get's much more messages and hence much less insertions into the QListWidget.

**Pull request merged**